### DISCUSSION TITLE
Create interactions between ligand and protein

### DISCUSSION THREAD

**Discussion by scipio095** [molstar/molstar#572] [Q&A]
Hello everyone, 

I have a pdb file and a ligand sdf loaded in molstar and used thanks to the molstar lib.
I have a tool who calculates interactions between ligand and protein and would like to know if it was possible to custom interactions in molstar by passing atom indexes in props and interaction type.

Thanks you for answers !

**Comment by dsehnal** [comment]
There is an example in src/examples which shows how to do that.

**Reply by scipio095** [reply]
Unfortunaly, we did not find the answers in examples.
We have already build a merged structure to merge the protein structure and the ligand structure and we can query a specific atom.
But we would like to know if it was possible to build a "non covalent interactions" programmatically by specifying the atoms and the types of interactions

**Reply by dsehnal** [reply]
Right, I see.

There are 2 options here:

- Build your own representation that shows the interactions (I've done this before in a proprietary solution)
- Extend the built-in representation to accept custom interaction data. We've talked with @arose about enabling this, but I am not sure either one of us will have time to implement this in the foreseeable future.

**Reply by SirBigoo** [reply]
Ok, so in this case we have to create a custom `StructureRepresentationProvider` as it is done for `InteractionsRepresentationProvider` ?

It takes some datas an input and I suppose it goes along and shows interactions. But, how to create an interaction on a structure ?

**Reply by dsehnal** [reply]
> But, how to create an interaction on a structure?

You mean how to pass down the data to the representation? For example by creating a custom property on the model. The data from there could then be used to obtain pairs (or even sets) of atoms to create the visuals from.

**Reply by dsehnal** [reply]
Btw it doesn't have to be a `StructureRepresentation`. It can be any custom shape visual to which you just pass the data required to constuct the interaction visuals.

**Reply by scipio095** [reply]
I found in the docking-viewer example the InteractionPreset example which show me how to build a structure with interactions computed by molstar (thanks to the selection query).

I have a contacts dictionnary with the type of interaction as key and an object as value which gives the atom concerned : 

Example : 
```json
{ 
'Hydrophobic' : {
 'at_l': 'F22',
'at_p': 'N'
} 
```

Now my question is how to apply these data to the merged structure ? I can have all the atoms from my protein and ligand, I just need to connect them.
I tried to override the computeInteractions method without success..

Hope I was clear enough..

**Reply by xiaoxudoo** [reply]
@scipio095 did u solive this? if yes, could u please share some code?
